From e625e8ec78fe3642769a5555c569b6298676acf4 Mon Sep 17 00:00:00 2001
From: Yuri Tolstov <yuri.tolstov@cavium.com>
Date: Sat, 18 Nov 2017 09:19:30 -0800
Subject: [PATCH 0364/1051] octeontx-bgx: Reduce BGX TX FIFO threshold

Due to big packet transfer latency jitter in PKO, the BGX TX FIFO
threshold should not be higher than PKO TX FIFO, which is for XLAUI port
is 10 KB. So, the threshold for this port is 8192 bytes.

Signed-off-by: Yuri Tolstov <yuri.tolstov@cavium.com>
Signed-off-by: Slawomir Rosek <slawomir.rosek@cavium.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/bgx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c b/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c
index bd86e1bff691..57d90bfe2fef 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c
@@ -213,7 +213,7 @@ static int bgx_port_initial_config(struct octtx_bgx_port *port)
 	switch (port->lmac_type) {
 	case OCTTX_BGX_LMAC_TYPE_40GR:
 		reg = 0x400;
-		thr = 0x300;
+		thr = 0x200;
 		break;
 	case OCTTX_BGX_LMAC_TYPE_XAUI:
 	case OCTTX_BGX_LMAC_TYPE_RXAUI:
-- 
2.17.1

